core/trino-main/src/main/java/io/trino/server/remotetask/HttpRemoteTask.java

```diff
@@ -581,71 +581,73 @@ private void triggerUpdate()
 
     private synchronized void sendUpdate()
     {
-        TaskStatus taskStatus = getTaskStatus();
-        // don't update if the task hasn't been started yet or if it is already finished
-        if (!started.get() || taskStatus.getState().isDone()) {
-            return;
-        }
-
-        int currentPendingRequestsCounter = pendingRequestsCounter.get();
-        if (currentPendingRequestsCounter == 0) {
-            return;
-        }
+        synchronized (this) {
```

Comment:
`Add explicit synchronization section` - you can drop the `synchronized` from method alrady.

Reply:
no, `sendUpdate` might still be scheduled twice or called from request callback. Only one should execute at same time.